Automattic/jetpack
# Premium Content Block: Move the product management system into a shared component

Fixes #23412
Fixes #23444

#### Changes proposed in this Pull Request:

* Premium Content Block: Move the product management system into a shared component.

This PR moves the Premium Content block's product-related toolbar and sidebar controls.
The plan is to eventually reuse it on several payment blocks (Premium Content and Payment Button), and to enhance it with more features.

The PR also moves the products into a Redux state, so that we'll be able to share them across several blocks, instead of fetching them every time.

The PR doesn't introduce any visual changes.
I've avoided attempting other improvements to avoid bloating this PR size.
Anything that we'll need to make this work on the Payment Button block (e.g. editorial changes), we'll do in a follow-up.

Note:
This could also include https://github.com/Automattic/jetpack/pull/23352.
I've chosen to not add it here as that introduces a new event that is a bit time-sensitive.
We can refactor the shared Stripe Connect button into this new component at a later time.

#### Jetpack product discussion

paYJgx-1SL-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:






* Sync these changes to WordPress.com, and test the Premium Content block with both free and business sites.
* On the free site: make sure the block displays the upgrade nudge and doesn't show any product-related UI.
* On the business site...
  * Stripe not connected?
    * Only the "Connect Stripe" control, no product-related UI.
  * No products (check `https://wordpress.com/earn/payments-plans/`)?
    * The block is automatically created with a $5/month subscription called "Monthly Subscription".
  * Click on the toolbar control:
    * Changing the selected product updates the `selectedPlanId` attribute of the block.
    * Clicking on "Add a new subscription" opens the block sidebar and focus on the name field.
  * From the sidebar:
    * Try inserting various subscriptions and make sure they get selected once inserted.
